src/test_stack.py

```diff
@@ -0,0 +1,81 @@
+"""Test stack.py"""
+
+import pytest
+
+
+@pytest.fixture
+def new_stack():
+    """Create a new stack with some data as a fixture."""
+    from stack import Stack
+    this_stack = Stack(data=[1, 2, 3])
+    return this_stack
+
+
+def test_new_stack_is_empty():
+    """Check that a new stack is empty."""
+    from stack import Stack
+    this_stack = Stack()
+    assert this_stack._linkedlist.size() == 0
+
+
+def test_new_stack_with_data_has_data(new_stack):
+    """Check new stack with has data."""
+    assert new_stack._linkedlist.head.value == 3
+
+
+def test_new_stack_with_head_has_head():
+    """Check new stack with head kwarg has head."""
+    from stack import Stack
+    from linked_list import Node
+    node1 = Node(5)
+    this_stack = Stack(head=node1)
+    assert this_stack._linkedlist.head.value == 5
+
+
+def test_pushed_value_is_new_head(new_stack):
+    """When pushing to stack, the pushed value should be the new head."""
+    new_stack._linkedlist.push("new")
+    assert new_stack._linkedlist.head.value == "new"
+
+
+def test_length_of_stack(new_stack):
+    """Check that size of the stack reflects the number of nodes in the stack."""
+    assert new_stack._linkedlist.length == 3
+
+
+def test_length_of_stack_when_pushed(new_stack):
+    """Check that size changes to reflect pushed values."""
+    new_stack._linkedlist.push("new")
```

Comment:
For this and a lot of similar following tests, you should be testing the Stack methods, not directly accessing the stack's `_linked_list` attribute.

This is why coverage for your `stack.py` file only shows as 54%